File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit;
        int min_value = INT_MAX;
        for (auto& it : prices) {
            min_value = min_value < it ? min_value : it;
            profit = profit < it - min_value ? it - min_value : profit;
        }
        return profit;
    }
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit=0;
        int min_value = Integer.MAX_VALUE;
        for(int i = 0; i< prices.length ; i++){
            min_value = min_value < prices[i] ? min_value : prices[i];
            profit = profit < prices[i] - min_value ? prices[i] - min_value : profit;
        }
        return profit;
    }
}